core/arch/aarchxx/mangle.c

```diff
@@ -268,18 +282,42 @@ insert_save_or_restore_registers(dcontext_t *dcontext, instrlist_t *ilist, instr
         if (reg1 == UINT_MAX)
             reg1 = i;
         else {
-            opnd_t mem = create_base_disp_for_save_restore(base_reg,
-                                                           false /* is_single_reg */,
-                                                           is_gpr, saved_regs);
-            if (save) {
-                new_instr = INSTR_CREATE_stp(dcontext, mem,
-                                             opnd_create_reg(first_reg + reg1),
-                                             opnd_create_reg(first_reg + i));
+            opnd_t mem1 = get_mem_opnd(base_reg, false /* is_single_reg */, is_gpr,
+                                       /* When creating save/restore instructions
+                                        * for inlining, we need the register id
+                                        * to compute the address.
+                                        */
+                                       ci != NULL ? first_reg + (reg_id_t)reg1
+                                                  : saved_regs, ci);
+
+            uint disp = opnd_get_disp(mem1);
+            /* We cannot use STP/LDP if the immediate offset is too big. */
+            if (disp > 504) {
```

Comment:
style: No magic constants: give it a name (and a comment as to where it comes from: 504 looks really odd)

Reply:
I've added a define + comment. Details: http://infocenter.arm.com/help/topic/com.arm.doc.dui0802a/STP_gen.html